### PR TITLE
[bug] lightning default dir integration fix

### LINKED ISSUE #361: `DVCLiveLogger` fails without `dir` value
The below snippet from https://dvc.org/doc/dvclive/get-started?tab=Pytorch-Lightning

```python
# train.py
from dvclive.lightning import DVCLiveLogger

...

trainer = Trainer(logger=DVCLiveLogger())
trainer.fit(model)
```

fails with:
>   File "/home/vscode/.local/lib/python3.10/site-packages/pytorch_lightning/loggers/logger.py", line 51, in get_experiment
>     return fn(self)
>   File "/home/vscode/.local/lib/python3.10/site-packages/dvclive/lightning.py", line 53, in experiment
>     self._experiment = Live(**self._live_init)
>   File "/home/vscode/.local/lib/python3.10/site-packages/dvclive/live.py", line 74, in __init__
>     self._init_paths()
>   File "/home/vscode/.local/lib/python3.10/site-packages/dvclive/live.py", line 115, in _init_paths
>     os.makedirs(self.dir, exist_ok=True)
>   File "/usr/local/lib/python3.10/os.py", line 210, in makedirs
>     head, tail = path.split(name)
>   File "/usr/local/lib/python3.10/posixpath.py", line 103, in split
>     p = os.fspath(p)
> TypeError: expected str, bytes or os.PathLike object, not NoneType
> Epoch 0: 100%|██████████| 11/11 [00:19<00:00,  1.77s/it, loss=0.181, v_num=_run]                                                           
> ERROR: failed to reproduce 'train': failed to run: python src/stages/train.py --config params.yaml, exited with 1

```
$ dvc doctor
DVC version: 2.34.0 (pip)
---------------------------------
Platform: Python 3.10.8 on Linux-5.4.0-1089-azure-x86_64-with-glibc2.31
Subprojects:
        dvc_data = 0.25.3
        dvc_objects = 0.12.2
        dvc_render = 0.0.12
        dvc_task = 0.1.4
        dvclive = 1.0
        scmrepo = 0.1.3
Supports:
        azure (adlfs = 2022.10.0, knack = 0.10.0, azure-identity = 1.11.0),
        gdrive (pydrive2 = 1.14.0),
        gs (gcsfs = 2022.10.0),
        hdfs (fsspec = 2022.10.0, pyarrow = 10.0.0),
        http (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        oss (ossfs = 2021.8.0),
        s3 (s3fs = 2022.10.0, boto3 = 1.24.59),
        ssh (sshfs = 2022.6.0),
        webdav (webdav4 = 0.9.8),
        webdavs (webdav4 = 0.9.8),
        webhdfs (fsspec = 2022.10.0)
Cache types: hardlink, symlink
Cache directory: ext4 on /dev/loop0
Caches: local
Remotes: s3
Workspace directory: ext4 on /dev/loop0
Repo: dvc, git
```

It works when `dir` is set in `DVCLiveLogger` i.e. `DVCLiveLogger(dir='my_logs_dir')`

### PR DESCRIPTION
**Pull request by shcheklein**
Fixes #361 

* [X] ❗ I have followed the [Contributing to DVCLive](https://github.com/iterative/dvclive/blob/master/CONTRIBUTING.md) guide.

* [X] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: src/dvclive/lightning.py [hunk 1]
```diff
@@ -20,10 +20,9 @@ def __init__(
 
         super().__init__()
         self._prefix = prefix
-        self._live_init = {
-            "dir": dir,
-            "resume": resume,
-        }
+        self._live_init = {"resume": resume}
```
Comment (skshetry) [inline src/dvclive/lightning.py:23]:
```suggestion
        self._live_init: Dict[str, Any] = {"resume": resume}
```

File: tests/test_lightning.py [hunk 2]
```diff
@@ -109,3 +109,18 @@ def test_lightning_integration(tmp_dir):
...
+    dvclive_logger = DVCLiveLogger("test_run")
+    trainer = Trainer(
+        logger=dvclive_logger,
+        max_epochs=2,
+        enable_checkpointing=False,
+        log_every_n_steps=1,
+    )
+    trainer.fit(model)
+
+    assert os.path.exists("dvclive")
```
Comment (skshetry) [inline tests/test_lightning.py:126]:
```suggestion
    assert (tmp_dir /"dvclive").is_dir()
```

File: src/dvclive/lightning.py (hunk 1 above)
Comment (shcheklein) [reply src/dvclive/lightning.py]: yep, fixed that already (mypy didn't run for some reason locally)

Comment (skshetry) [reply src/dvclive/lightning.py]: You need to use `nox -s lint` to run all of the linters. See https://github.com/iterative/dvclive/blob/main/CONTRIBUTING.rst#how-to-submit-changes.

File: tests/test_lightning.py (hunk 2 above)
Comment (shcheklein) [reply tests/test_lightning.py:126]: is it actually needed? some other tests should be fixed as well .. @skshetry please feel free to commit this

Comment (skshetry) [reply tests/test_lightning.py:126]: Let's leave it out then. We should use `pathlib.Path` when we can, as it comes from fixture itself.

**Review by skshetry — APPROVED** [review]

**Comment by skshetry** [comment]
@shcheklein, do we need to release? Let me know.  Also see https://github.com/iterative/dvc/wiki/Releasing-dvc*-projects.

**Pull request merged**